torch/lib/c10d/ProcessGroupGloo.hpp

```diff
@@ -265,7 +272,9 @@ class ProcessGroupGloo : public ProcessGroup {
   // Checkout constructs new AlgorithmEntry or returns existing one.
   AlgorithmEntry* checkout(const KeyType& key);
 
-  std::unordered_map<KeyType, EntryType, HashType> cache_;
+  const int cacheNumAlgorithmEntries_;
+  std::unordered_map<KeyType, int, HashType> cacheCurrentEntry_;
```

Comment:
can you add a comment on what these two hash tables do

Reply:
Good point, adding it.